qiskit/pulse/transforms.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Basic rescheduling functions which take schedules or instructions
+(and possibly some arguments) and return new schedules.
+"""
+from collections import defaultdict
+
+from . import Play
+from .schedule import Schedule
+
+
+def compress_pulses(schedule: Schedule) -> Schedule:
+    """Optimization pass to replace identical pulses.
+
+    Args:
+        schedule: Schedule to compress.
+
+    Returns:
+        Compressed schedule.
+    """
+
+    new_schedule = Schedule(name=schedule.name)
+    existing_pulses = defaultdict(list)
+
+    for time, inst in schedule.instructions:
+        if isinstance(inst, Play):
+            existing_channel_pulses = existing_pulses[inst.channel.name]
```

Comment:
I don't think this comparison has to be by channel. It could be for all pulses in the schedule. Maybe we could have a transform doing this by channel or for all channels?

Reply:
added `by_channel` parameter to transformer and set default to `False`, so compression applied to all channels of `Schedule`.
https://github.com/Qiskit/qiskit-terra/pull/4114/commits/5c504610b31cec68304fc0ec77c9f17f2257c49f#diff-6d98eda127a5a63ebe9c65e02bf6bf35R42